### ISSUE TITLE
Circe fails to find decoders

### ISSUE THREAD

**Issue by threeseed** [shogowada/scala-json-rpc#41]
Getting the following errors trying to use Circe:

```
could not find implicit value for parameter decoder: io.circe.Decoder[io.github.shogowada.scala.jsonrpc.Models.JSONRPCId]

could not find implicit value for parameter decoder: io.circe.Decoder[io.github.shogowada.scala.jsonrpc.Models.JSONRPCRequest[(com.baank.models.Page, String)]]
```

**Comment by shogowada** [comment]
It's probably the same issue as https://github.com/shogowada/scala-json-rpc/issues/42; Circe cannot tell how to serialize `com.baank.models.Page`. Please make sure Circe can serialize `com.baank.models.Page` without scala-json-rpc.

**Comment by threeseed** [comment]
I have removed the bindAPI entirely. So there is no link between scala-json-rpc and my code. I also tried using a new API class with empty methods to make sure it wasn't a serialisation issue with my code. I get this error:

could not find implicit value for parameter decoder: io.circe.Decoder[io.github.shogowada.scala.jsonrpc.Models.JSONRPCId]
[error] 			serverAndClient.receiveAndSend(message).onComplete {

Still seems like it's failing to serialize JSONRPCId.
